clusterlib/tests/test_scheduler.py

```diff
@@ -4,19 +4,40 @@
 from __future__ import unicode_literals
 
 import os
+import os.path as op
+from time import sleep
 import subprocess
 from getpass import getuser
-from xml.etree.ElementTree import XMLParser, XML
 
 from nose.tools import assert_equal
 from nose.tools import assert_raises
 from nose.tools import assert_in
 from nose import SkipTest
 
-from ..scheduler import queued_or_running_jobs
-from ..scheduler import submit
-from ..scheduler import _which
-from ..scheduler import _get_backend
+from clusterlib.scheduler import queued_or_running_jobs
+from clusterlib.scheduler import submit
+from clusterlib.scheduler import _which
+from clusterlib.scheduler import _get_backend
+from clusterlib.testing import TemporaryDirectory
+
+
+def _do_dispatch(*args, **kwargs):
+    """Perform a dispatch with the submit command and return the job id"""
+    # TODO: This utility function should be properly documented any made more
+    # robust to be included in the scheduler module itself
```

Comment:
It's more robust to get the `job_id` from the output of qsub / sbatch directly that querying with the job name. The job name is not guaranteed to be unique (if you have two instances of the same dispatch script running concurrently for instance).

I will need to get the exact job id at submission time to resolve race conditions in the futures branch.